tests/test_http_ext_auth.py

```diff
@@ -1417,8 +1417,8 @@ async def test_http_auth_ext_azure_authorize_01(self):
 
             discovery_request = (
                 "GET",
-                "https://login.microsoftonline.com/common/v2.0",
-                ".well-known/openid-configuration",
+                "https://login.microsoftonline.com",
+                "common/v2.0/.well-known/openid-configuration",
```

Comment:
Why do these need to change?

Reply:
This was relying on some very obscure parsing and encoding for base URLs. In the end I just cleaned up the URL handling so we didn't have to special case the base URL part of things.